Expanded on Disabled Hails (Fixed)

OverYrPaygrade opened the pull request:
I screwed up the previous request due to technical difficulties and decided it was easier to close the pull and try again.

This PR expands on the list of hails a disabled friendly ship can say when contacted.

This one fixes the capitalization and spelling errors.

Bladewood commented:
From now on, use GitHub's branching system to avoid errors, as in #3718.
Create a new branch on your repository, and push your desired changes to that branch instead of to the master branch. Then you won't break stuff when a new commit is pushed to Michael's endless-sky repository or such.

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,273 @@ phrase "friendly author"
+		"You answered! "
+		"Good to see a friendly face! "
+		""
+		"Hey, pleaase! "
```
double a in please
and why not put all "" (empty words) after each other instead of spread out?

Bladewood replied:
Because we have three `""`s here already. If these spaces were moved into its own separate word, we could end up with
`" Our ship is disabled, could you lend a hand?"` with an extra space at the front.

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,273 @@ phrase "friendly author"
+		"could you render assistance"
+		"can we get a little help here"
+	word
+		"?"
```
same here, 3 versions of this word are "?" but they aren't put together
not really a big issue but organize them

Bladewood commented on data/hails.txt:
```diff
@@ -2030,7 +2030,273 @@ phrase "friendly author"
+		" so we can get out of here?"
+		" as soon as possible?"
+		"?"
+		"?"
```
There should be a newline from here to the next phrase.
It would also be cleaner faster to move all of these question marks:
```
    word
        "?"
        "?"
        "?"
        "?"
        "?"
        "?"
        "? I don't wanna get stranded here!"

Bladewood commented on data/hails.txt:
```diff
@@ -2030,7 +2030,273 @@ phrase "friendly author"
+		" disabled us! "
+		" fried most of our ship functions! "
+		" suddenly came onto us! "
+		" thought they could just leave us here! "
```
It would be better here to put a space-word in front and behind.

Bladewood commented on data/hails.txt:
```diff
@@ -2030,7 +2030,273 @@ phrase "friendly author"
+		" as soon as possible?"
+		"?"
+		"?"
+phrase "friendly disabled"
```
Same newline issue.

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2045,23 +2047,27 @@ phrase "friendly disabled"
-		"We're stuck drifting around in this deathtrap, "
-		"Not to be a bother, but "
-		"Could you come over, and "
+		" "
```
the first word has some "" versions, which means such hails "start" at the 3rd word, while these would get this empty space before them
remove this word entirely and add a space after the non-empty versions of word 1

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2045,23 +2047,27 @@ phrase "friendly disabled"
+		"You're a friendly, right?"
+		"You answered!"
+		"Good to see a friendly face!"
+		"Hey, please!"
```
from my other comment, these ones would get the space after them

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+	word
+		"Please come board our ship and patch us up!"
+		"We would appreciate any help getting us running again!"
+		"We can't fix this on our own!"
```
not really sure if this line is... good (as in, it sounds impolite in some combinations)
imagine "Hey! We can't fix this on our own!", and technically they don't even ask for help at all
personally I'd remove this line

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"Please come board our ship and patch us up!"
+		"We would appreciate any help getting us running again!"
+		"We can't fix this on our own!"
+		"Could you fly by and help us please?"
```
I'd say put a comma after "us", to get "help us, please?"

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"We're in a bit of a bind here,"
+		"We're stuck drifting around in this deathtrap,"
+		"Not to be a bother, but"
+		"Could you come over, and"
```
this doesn't really work with several combinations
"Could you come over, and [could you lend a hand / ccan you get us back on our feet / would you help us" etc.
basically they ask if you can come over, and they ask for repairs
the rest of the variants of this word seem to work, so removing this one seems to be the easiest way

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"could you render assistance"
+		"can we receive some help"
+	word
+		" please?"
```
I think it's just personal preference, not sure, but I'd put a comma here too, to make ", please?"

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"can we get some help paching up our ship"
+	word
+		" please?"
+		" please?"
```
in case you agree to put the commas before please's, here's 2 more

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"got the jump on us!"
+		"left us here to die!"
+		"disabled us!"
+		"fried most of our ship functions!"
```
I think it should be ship's functions

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"Could you lend a hand?"
+		"We can't seem to repair it ourselves, could you help?"
+		"We need help getting our ship running again!"
+		"Can you help us fix it please?"
```
another please that [should? could?] be comma'd
the next one shouldn't, in case you wonder, because please is before the rest of the sentence

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"Please help us!"
+		"I don't think we can hold out!"
+		"Please come and help!"
+		"Any chance you can get us out of a bind?"
```
looks a bit strange, I'd say change "a" to "this"

KiLEdEnNis commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"Please come and help!"
+		"Any chance you can get us out of a bind?"
+		"Any chance you could bail us out?"
+		"We need you to come over to patch us up!"
```
sounds a little rude

Bladewood commented on data/hails.txt:
```diff
@@ -2030,7 +2030,288 @@ phrase "friendly author"
+		"would you help us"
+		"is there a chance you could get us up and running again"
+		"could you render assistance"
+		"can we get some help paching up our ship"
```
patching

OverYrPaygrade commented:
Your suggestions made a lot of sense. Adjusted the hails to account for everything you said.

CyberShadow commented:
Reposting [my comment](https://github.com/endless-sky/endless-sky/pull/3718#issuecomment-385182059) from the previous PR:

If the current message didn't explicitly mention boarding, I don't think I would have figured out how to actually help them. I think that every message ought to mention boarding somehow, or a help text could be accompanied with the first time such a message is seen by the player.

OverYrPaygrade commented:
Having a help message when selecting a friendly disabled ship for the first time would be helpful for what you're saying.
I'm not quite sure how to do that, though. I think that's a coding issue..?

Though if I did figure it out I would make this text for it.
`help "friendly disabled"
	This ship is disabled and needs your help patching them up! Pressing <boarding> will fly you to their ship and get them up and running again. Just be sure to do when no one is trying to kill you, or they could get caught in the crossfire!`

CyberShadow commented:
@OverYrPaygrade Sent you a PR: https://github.com/OverYrPaygrade/endless-sky/pull/1

OverYrPaygrade commented:
Thanks to Cyber for coding in functionality for help text for friendly disabled ships. Since a lot of the hails I wrote don't mention how to repair disabled ships, this should help players with how to do so.

MCOfficer commented on data/outfits.txt:
```diff
@@ -396,3 +396,15 @@ outfit "Militia Carrier License"
+	"heat generation" .7
+	description "Not all upgrades to your ship have to be purely functional. The addition of a shower utilizing actual hot water may not kill your enemies, provide power to your engines, or shield you from a missile barrage, but it makes life aboard a ship a lot more comfortable. Captains who spend most of their time in space and not planetside will definately appreciate this addition to their vesse ...
+
+    The morale benefits of having one installed are more than enough to convince those with the money to spend on a little luxury to get one installed in their ship. Plus, they are particularly attractive to rich passengers willing to pay top dollar for a comfortable ship to seek passage on."
```
This should not be part of this PR.  You might want to create a new branch for these changes, and for future PRs don't compare from your master, but use a separate branch instead.

KiLEdEnNis commented:
the shakespearean insults can become a phrase (think 2, adjective & noun), and both the hostile and friendly disabled hails could use this list
like how names.txt has phrase "digit" and uses this for "syndicate small"

it'd save some lines

OverYrPaygrade commented:
I have two options at the ready.

One is to put the shakespearean insults into phrases that are referenced by both hail sets. To be compatible with the friendly disabled hails, I changed the 3rd set to be plural, and adjusted the hostile disabled hail to make sense grammatically in plural. Makes sense, as you're often flying a ship with a crew complement, and even if you aren't the person could be yelling at you in general. Only the second and third parts of the insults are used for the friendly hails, and all three for the hostile hail.

However, if MZ would rather a friendly hail that refrains from using the insult list, I can make a simplified friendly phrase using more common insult wording for the hail.

endless-sky commented:
My preference is for the Shakespearean insults to only be used by hostile ships.

KiLEdEnNis commented:
to be honest, so is mine; seeing them on hostile disableds is just some funny joke, but it feels a bit too much if friendlies use it too, especially if the friendly disabled hails are way longer (several sentences)

OverYrPaygrade commented:
Okay. This version takes away the shakespearean lists and substitutes a small assortment of other insults.
